ntoskrnl/include/internal/mm.h

```diff
@@ -866,6 +866,34 @@ NTAPI
 MmDeleteKernelStack(PVOID Stack,
                     BOOLEAN GuiStack);
 
+/* balance.c / pagefile.c******************************************************/
+
+inline VOID UpdateTotalCommittedPages(LONG Delta)
+{
+    /*
+     * Add up all the used "Committed" memory + pagefile.
+     * Not sure this is right. 8^\
+     * MmTotalCommittedPages should be adjusted consistently with
+     * other counters at different places.
+         MmTotalCommittedPages = MiMemoryConsumers[MC_SYSTEM].PagesUsed +
+                            MiMemoryConsumers[MC_USER].PagesUsed +
+                            MiUsedSwapPages;
+     
+     */
+    
+    /* Update Commitment */
+    SIZE_T TotalCommittedPages = InterlockedExchangeAddSizeT(&MmTotalCommittedPages, Delta) + Delta;
+
+    SIZE_T PeakCommitment = MmPeakCommitment;
+
+    /* Update Peak */
+    while (TotalCommittedPages > PeakCommitment &&
+           InterlockedCompareExchangeSizeT(&MmPeakCommitment, TotalCommittedPages, PeakCommitment) != PeakCommitment)
+    {
+        PeakCommitment = MmPeakCommitment;
+    }
```

Comment:
Side remark: an explanation of this pattern can be found e.g. at
https://devblogs.microsoft.com/oldnewthing/20040915-00/?p=37863

Reply:
Do you want me to add this as comment in code ?

Reply:
No, it was just for your (and possible other readers) personal knowledge.

Reply:
Ok, thanks!